cupy/testing/helper.py

```diff
@@ -101,8 +101,13 @@ def test_func(self, *args, **kw):
             # values are negative.
             if _contains_signed_and_unsigned(kw):
                 inds = _make_positive_indices(self, impl, args, kw)
-                cupy_result = cupy.asnumpy(cupy_result)[inds]
-                numpy_result = cupy.asnumpy(numpy_result)[inds]
+                try:
```

Comment:
I don't think that the referenced code do what it should do, because it skips the test.
I think it is better to have a fallback for the index error, instead to skip check_func.